samples/utils/src/main/java/quarks/samples/utils/sensor/SimulatedTemperatureSensor.java

```diff
@@ -0,0 +1,115 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+
+package quarks.samples.utils.sensor;
+
+import java.text.DecimalFormat;
+import java.util.Objects;
+import java.util.Random;
+
+import quarks.analytics.sensors.Range;
+import quarks.analytics.sensors.Ranges;
+import quarks.function.Supplier;
+
+/**
+ * A Simulated temperature sensor.
+ * <p>
+ * The sensor starts off with an initial value.
+ * Each call to {@link #get()} changes the current value by
+ * a random amount between plus/minus a {@code deltaFactor}.
+ * The new current value is limited to a {@code maxTempRange}.
+ * <p>
+ * No temperature scale is implied (e.g., Fahrenheit, Kelvin, ...).
+ * The {@code double} temperature values are simply generated as described.
+ * The user of the class decides how to interpret them.
+ * <p>
+ * Sample use:
+ * <pre>{@code
+ * Topology t = ...;
+ * SimulatedTemperatureSensor tempSensor = new SimulatedTemperatureSensor();
+ * TStream<Double> temp = t.poll(tempSensor, 1, TimeUnit.SECONDS);
+ * }</pre>
+ */
+public class SimulatedTemperatureSensor implements Supplier<Double> {
+    private static final long serialVersionUID = 1L;
+    private static DecimalFormat df = new DecimalFormat("#.#");
+    private Random r = new Random();
+    private final Range<Double> maxTempRange;
+    private final double deltaFactor;
+    private double currentTemp;
+   
+    /**
+     * Create a temperature sensor.
+     * <p>
```

Comment:
It's a good habit in javadoc to close opened `<P>` with `</P>`.

Some HTML verification/sccessibility testing tools will complain about paragraphs that are not closed.